### PR TITLE
gtkcord4: install desktop item and icons, optional libadwaita

### PR DESCRIPTION
**Pull request by hmenke**
###### Description of changes

Upstream ships a desktop item and icons, so we might as well install them in the standard paths.
There is also an option to build against libadwaita, but upstream disables this by default, so we should too.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/audio/gtkcord4/default.nix
Comment (urandom2) [inline pkgs/applications/audio/gtkcord4/default.nix:14]: can you move this up one? alphabetically`wi` comes before `wr`

**Review by urandom2 — APPROVED** [review]

File: pkgs/applications/audio/gtkcord4/default.nix
Comment (urandom2) [inline pkgs/applications/audio/gtkcord4/default.nix:50]: while this does work, I think `$out` reads better:
```suggestion
    install -D -m 444 -t $out/share/applications .nix/com.github.diamondburned.gtkcord4.desktop
    install -D -m 444 internal/icons/svg/logo.svg $out/share/icons/hicolor/scalable/apps/gtkcord4.svg
    install -D -m 444 internal/icons/png/logo.png $out/share/icons/hicolor/256x256/apps/gtkcord4.png
```

**Review by urandom2 — APPROVED** [review]

**Comment by SuperSandro2000** [comment]
> There is also an option to build against libadwaita, but upstream disables this by default, so we should too.

Why though? Nix has no redistribution problems like when downloading the precompiled binary. If it works then it should just be enabled with no option.

**Comment by hmenke** [comment]
To quote the developer of gtkcord4:

> I don't want to use libadwaita anymore after they decided to force the theme on my application, even though I am the developer, not them. — https://github.com/diamondburned/gtkcord4/issues/24#issuecomment-1138757107

I think we should respect that decision, even though they were so generous as to continue maintaining libadwaita support.

**Review by urandom2 — APPROVED** [review]

**Comment by SuperSandro2000** [comment]
nixpkgs enables most optional features by default. Also most graphic libraries enforce lots of things for example through theming.

Are there any other problems? Is the only problem that end users require libadwaite to be installed? Then that would not apply to nixpkgs.

**Comment by RaitoBezarius** [comment]
> Are there any other problems? Is the only problem that end users require libadwaite to be installed? Then that would not apply to nixpkgs.

It's a policy issue, given the discussion in the issue mentioned above, I would tend to honor the upstream defaults as we  do not want extra maintenance cost which comes with the fact this program do not want to maintain libadwaita integration IMHO.

It's easier to say: it has a libadwaita integration which is known to not be used actively by developers, here be dragons. Rather than enable it by default and pouring bug reports, which could bring the developer to pull out this integration, etc.

I am in favor of this change and honoring upstream wishes here.

**Review by RaitoBezarius — APPROVED** [review]
Will not merge until enough maintainers express their opinion on this change.

**Pull request merged**
